networkconfig/api/v1alpha/networkconfig_types.go

```diff
@@ -0,0 +1,81 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// AvailableIPMapping groups an IP address and Name. This is a struct and not a Map for future extensibility
+type AvailableIPMapping struct {
+	Name string `json:"name,omitempty"`
+	IP   string `json:"ip,omitempty"`
+}
+
+// NetworkContainer defines the structure of a Network Container as found in NetworkConfigStatus
+type NetworkContainer struct {
+	PrimaryIP      string               `json:"primaryIP,omitempty"`
+	ID             string               `json:"id,omitempty"`
+	Subnet         string               `json:"subnet,omitempty"`
+	AvailableIPs   []AvailableIPMapping `json:"availableIPs,omitempty"`
+	IPCount        int64                `json:"ipCount,omitempty"`
+	DefaultGateway string               `json:"defaultGateway,omitempty"`
+	Netmask        string               `json:"netmask,omitempty"`
+}
+
+// NetworkConfigSpec defines the desired state of NetworkConfig
+type NetworkConfigSpec struct {
+	IPCount     int64    `json:"ipCount,omitempty"`
```

Comment:
My 2 cents, Can we rename it as RequestedIPCount? It will be more explicit. IPCount may still confuse as a reference count to the IPsNotInUse list.